core/base/planarGraphLayout/MergeTreeVisualization.h

```diff
@@ -0,0 +1,930 @@
+/// \ingroup base
+/// \class MergeTreeVisualization
+/// \author Mathieu Pont (mathieu.pont@lip6.fr)
+/// \date 2021.
+///
+/// Visualization module for merge trees.
+///
+
+#ifndef _MERGETREEVISUALIZATION_H
+#define _MERGETREEVISUALIZATION_H
+
+#pragma once
+
+#include <FTMTree.h>
+
+#include <stack>
+
+using namespace ttk;
+using namespace ftm;
+
+class MergeTreeVisualization : virtual public Debug {
+protected:
+  // Visualization parameters
+  bool PlanarLayout = false;
+  bool BranchDecompositionPlanarLayout = false;
+  double BranchSpacing = 1.;
+  bool RescaleTreesIndividually = false;
+  double ImportantPairs = 50.; // important pairs threshold
+  double ImportantPairsSpacing = 1.;
+  double NonImportantPairsSpacing = 1.;
+  double NonImportantPairsProximity = 0.05;
+
+public:
+  MergeTreeVisualization(){};
+  ~MergeTreeVisualization(){};
+
+  // ==========================================================================
+  // Getter / Setter
+  // ==========================================================================
+  // Visualization parameters
+  void setPlanarLayout(bool b) {
+    PlanarLayout = b;
+  }
+  void setBranchDecompositionPlanarLayout(bool b) {
+    BranchDecompositionPlanarLayout = b;
+  }
+  void setBranchSpacing(double d) {
+    BranchSpacing = d;
+  }
+  void setRescaleTreesIndividually(bool b) {
+    RescaleTreesIndividually = b;
+  }
+  void setImportantPairs(double d) {
+    ImportantPairs = d;
+  }
+  void setImportantPairsSpacing(double d) {
+    ImportantPairsSpacing = d;
+  }
+  void setNonImportantPairsSpacing(double d) {
+    NonImportantPairsSpacing = d;
+  }
+  void setNonImportantPairsProximity(double d) {
+    NonImportantPairsProximity = d;
+  }
+
+  // ==========================================================================
+  // Planar Layout
+  // ==========================================================================
+  // TODO manage multi pers pairs
+  template <class dataType>
+  void treePlanarLayoutBDImpl(
+    FTMTree_MT *tree,
+    std::vector<float> &retVec,
```

Comment:
what are these `float` about ?

Reply:
This vector of float contains the coordinates in 2D of the nodes of a merge tree, it is filled by the planar layout algorithm.

Reply:
So we have `[x1, y1, x2, y2, ...]` in a flat array ? If so, having a `using ptCoords = tuple<float,float>` and a `vector<ptCoords>` would have been great.
Can we use a more explicit name like nodeCoords ?